frontend/src/app/components/pool/pool.component.ts

```diff
@@ -0,0 +1,85 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+import { once } from 'process';
+import { BehaviorSubject, combineLatest, from, merge, Observable } from 'rxjs';
+import { delay, distinctUntilChanged, map, scan, startWith, switchMap, tap } from 'rxjs/operators';
+import { BlockExtended, PoolStat } from 'src/app/interfaces/node-api.interface';
+import { ApiService } from 'src/app/services/api.service';
+import { StateService } from 'src/app/services/state.service';
+
+@Component({
+  selector: 'app-pool',
+  templateUrl: './pool.component.html',
+  styleUrls: ['./pool.component.scss']
+})
+export class PoolComponent implements OnInit {
+  poolStats$: Observable<PoolStat>;
+  blocks$: Observable<BlockExtended[]>;
+
+  fromHeight: number = -1;
+  fromHeightSubject: BehaviorSubject<number> = new BehaviorSubject(this.fromHeight);
+
+  blocks: BlockExtended[] = [];
+  poolId: number = undefined;
+  isLoading = false;
+  radioGroupForm: FormGroup;
+
+  constructor(
+    private apiService: ApiService,
+    private route: ActivatedRoute,
+    public stateService: StateService,
+    private formBuilder: FormBuilder,
+  ) {
+    this.radioGroupForm = this.formBuilder.group({ dateSpan: '1w' });
+    this.radioGroupForm.controls.dateSpan.setValue('1w');
+  }
+
+  ngOnInit(): void {
+    this.poolStats$ = combineLatest([
+      this.route.params.pipe(map((params) => params.poolId)),
+      this.radioGroupForm.get('dateSpan').valueChanges.pipe(startWith('1w')),
+    ])
+      .pipe(
+        switchMap((params: any) => {
+          this.poolId = params[0];
+          if (this.blocks.length === 0) {
+            this.fromHeightSubject.next(undefined);
+          }
+          return this.apiService.getPoolStats$(this.poolId, params[1] ?? '1w');
+        }),
+        map((poolStats) => {
+          let regexes = '"';
+          for (const regex of JSON.parse(poolStats.pool.regexes)) {
```

Comment:
Move the JSON.parse( to the backend?

Reply:
Fixed in https://github.com/mempool/mempool/pull/1242/commits/fa8607c57d30880abcb17b8a5ed69db462da7012